ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/ObserverWithResult.scala

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.benchtool.metrics
+
+import io.grpc.stub.StreamObserver
+import org.slf4j.Logger
+
+import scala.concurrent.{Future, Promise}
+
+abstract class ObserverWithResult[T](logger: Logger) extends StreamObserver[T] {
+
+  def streamName: String
+
+  def result: Future[Unit] = promise.future
+
+  private val promise = Promise[Unit]()
+
+  override def onNext(value: T): Unit = {
+    ()
+  }
+
+  override def onError(t: Throwable): Unit = {
+    logger.error(withStreamName(s"Received error: $t"))
+    promise.failure(t)
+  }
+
+  override def onCompleted(): Unit = {
+    logger.debug(withStreamName(s"Completed."))
```

Comment:
You used to log completion at an info level. I would keep one message at the start and one at the end of a stream with that level.